docs/framework/react/guides/advanced-ssr.md

```diff
@@ -532,6 +532,25 @@ Now, your `getPosts` function can return e.g. `Temporal` datetime objects and th
 
 For more information, check out the [Next.js App with Prefetching Example](../examples/nextjs-app-prefetching).
 
+### Using the Persist Adapter with Streaming
+
+If you're using the persist adapter with the [Streaming with Server Components](#streaming-with-server-components) feature, you need to be careful not to save promises to storage. Since pending queries can be dehydrated and streamed to the client, you should configure the persister to only persist settled queries:
```

Comment:
```suggestion
If you're using the persist adapter with the [Streaming with Server Components](#streaming-with-server-components) feature, you need to be careful not to save promises to storage. Since pending queries can be dehydrated and streamed to the client, you should configure the persister to only persist successful queries:
```